vaadin-upload-flow-parent/vaadin-upload-flow/src/test/java/com/vaadin/flow/component/upload/UploadClearFilesTest.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright 2000-2021 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.vaadin.flow.component.upload;
+
+import static org.junit.Assert.*;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import com.vaadin.flow.component.UI;
+import com.vaadin.flow.server.VaadinRequest;
+import com.vaadin.flow.server.VaadinService;
+import com.vaadin.flow.server.VaadinServlet;
+import com.vaadin.flow.server.VaadinServletService;
+import com.vaadin.flow.server.VaadinSession;
+
+import elemental.json.Json;
+
+import java.io.IOException;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+public class UploadClearFilesTest {
+
+    String mockJson = "[{\"loaded\": 11037,\"held\": false,\"status\": \"\",\"xhr\": {},\"uploadTarget\": \"VAADIN/dynamic/resource/5/53877f5e-28d9-47a2-b061-80670615fb6e/upload\",\"formDataName\": \"file\",\"indeterminate\": false,\"uploading\": false,\"error\": false,\"abort\": false,\"complete\": true,\"progress\": 100}]";
+
+    private Upload upload;
+
+    @Before
+    public void initTest() throws IOException {
+        UI ui = new MockUI();
```

Comment:
I guess we can avoid adding `MockUI` and `MockVaadinSession` and instead create an integration test. (For example, modify `UploadView` and add test to `UploadIT`)